src/util/mapbox.js

```diff
@@ -389,12 +403,59 @@ class TelemetryEvent {
         });
     }
 
-    queueRequest(event: number | {id: number, timestamp: number}, customAccessToken?: ?string) {
+    queueRequest(event: any, customAccessToken?: ?string) {
         this.queue.push(event);
         this.processRequests(customAccessToken);
     }
 }
 
+export class PerformanceEvent extends TelemetryEvent {
+    errorCb: EventCallback;
+
+    constructor() {
+        super('gljs.performance');
+    }
+
+    postPerformanceEvent(customAccessToken: ?string, performanceData: LivePerformanceData, callback: EventCallback) {
+        this.errorCb = callback;
+
+        if (config.EVENTS_URL) {
+            if (customAccessToken || config.ACCESS_TOKEN) {
+                this.queueRequest({timestamp: Date.now(), performanceData}, customAccessToken);
+            } else {
+                this.errorCb(new Error(AUTH_ERR_MSG));
```

Comment:
We can skip here and not send the error as it's not actually used (Refer the internal empty callback function [here](https://github.com/mapbox/mapbox-gl-js/blob/karim/live-performance-counters/src/ui/map.js#L3294)).

I initially left it here for completion based on the pattern already used in other event methods. Since it's not propagated across the API surface back to the customer, we can pass and remove it.